suites/test_bulk_operations.py

```diff
@@ -0,0 +1,147 @@
+
+    @lcc.test('Verify user can bulk unpublish modules')
+    # @lcc.disabled()
+    def modules_bulk_unpublish(self):
+        poll(lambda: len(utilities.find_elements_by_css_selector(self.driver, locators.MODULES_LIST_CSS)) == 4,
+             ignore_exceptions=[NoSuchElementException],
+             timeout=5,
+             step=0.5)
+        utilities.click_element(self.driver, By.XPATH, locators.MODULES_SELECT_ALL_TITLE_XPATH)
+        # Deselect first title
+        utilities.click_element(self.driver, By.XPATH, locators.FIRST_MODULE_CHECKBOX_XPATH)
+        utilities.wait(2)
+        bulk_unpublish = utilities.find_element(self.driver, By.CSS_SELECTOR, locators.BULK_UNPUBLISH_CSS)
+        bulk_unpublish.click()
+        check_that("Confirmation modal title",
+                   utilities.get_text(self.driver, By.CSS_SELECTOR, locators.MODAL_TITLE_CSS),
+                   equal_to("Unpublish"))
+        check_that("Count of modules being unpublished",
+                   utilities.get_text(self.driver, By.CSS_SELECTOR, locators.TITLES_FOR_UNPUBLISH_CSS),
+                   contains_string("3"))
+        utilities.click_element(self.driver, By.CSS_SELECTOR, locators.CONFIRM_BUTTON_CSS)
+        poll(lambda: utilities.get_text(self.driver, By.CSS_SELECTOR, locators.PROGRESS_SUCCESS_STATUS) == "100%",
+             ignore_exceptions=[NoSuchElementException],
+             timeout=20,
+             step=1)
+        utilities.click_element(self.driver, By.XPATH, locators.VIEW_DETAILS_LINK)
+        modules_updated = utilities.find_elements_by_css_selector(self.driver, locators.UPDATED_TITLES_LIST_CSS)
+        check_that("Count of successfully updated modules", len(modules_updated), equal_to(3))
+        check_that("Failed modules section text",
+                   utilities.get_text(self.driver, By.CSS_SELECTOR, locators.FAILED_TITLES_LIST_CSS), equal_to("n/a"))
+        utilities.click_element(self.driver, By.XPATH, locators.CLOSE_DETAILS_XPATH)
+        utilities.click_element(self.driver, By.CSS_SELECTOR, locators.CLOSE_STATUS_ALERT)
+
+    @lcc.test('Verify cannot edit metadata for published document')
+    # @lcc.disabled()
+    def published_title_add_metadata(self):
+        utilities.click_element(self.driver, By.XPATH, locators.FIRST_MODULE_CHECKBOX_XPATH)
+        bulk_edit_metadata = utilities.find_element(self.driver, By.XPATH, locators.BULK_EDIT_METADATA_XPATH)
+        bulk_edit_metadata.click()
+        search_beta_page.add_bulk_metadata(self.driver)
+        check_that("Alert message displayed when trying to add metadata to published title",
+            utilities.get_text(self.driver, By.CSS_SELECTOR, locators.ALERT_TITLE_CSS),
+            equal_to(constants.error_message_on_edit_metadata))
+        utilities.click_element(self.driver, By.XPATH, locators.CLOSE_DETAILS_XPATH)
+        utilities.wait(15)
+        lcc.log_info("End of tests, executing teardown")
+
```

Comment:
@noopurmath3 please remove the extra blank lines. only one blank line is expected at the end of file. thanks.